x/sequencers/keeper/dymint.go

```diff
@@ -13,40 +13,59 @@ import (
 )
 
 // set dymint sequencers from InitChain
-func (k Keeper) SetDymintSequencers(ctx sdk.Context, validators []abci.ValidatorUpdate) {
-	for _, val := range validators {
-		tmkey, err := tmcrypto.PubKeyFromProto(val.PubKey)
+func (k Keeper) SetDymintSequencers(ctx sdk.Context, sequencers []abci.ValidatorUpdate) {
+	if len(sequencers) == 0 {
+		panic(types.ErrNoSequencerOnInitChain)
+	}
+
+	if len(sequencers) > 2 {
+		panic(types.ErrMultipleDymintSequencers)
+	}
+
+	var (
+		operatorAddr    sdk.ValAddress
+		power           int64
+		consensusPubKey cryptotypes.PubKey
+		operatorPubkey  cryptotypes.PubKey
+	)
+
+	for _, seq := range sequencers {
+		tmkey, err := tmcrypto.PubKeyFromProto(seq.PubKey)
 		if err != nil {
 			panic(err)
 		}
 		pubKey, err := cryptocodec.FromTmPubKeyInterface(tmkey)
 		if err != nil {
 			panic(err)
 		}
-		err = k.SetDymintSequencerByAddr(ctx, sdk.ConsAddress(pubKey.Address()), uint64(val.Power))
-		if err != nil {
-			panic(err)
+
+		if pubKey.Type() == ed25519.KeyType {
+			consensusPubKey = pubKey
+			power = seq.Power
+		} else {
+			operatorPubkey = pubKey
+			operatorAddr = sdk.ValAddress(pubKey.Address())
 		}
 	}
-}
 
-// get a single sequencer registered on dymint by consensus address
-func (k Keeper) GetDymintSequencerByAddr(ctx sdk.Context, consAddr sdk.ConsAddress) (power uint64, found bool) {
-	store := ctx.KVStore(k.storeKey)
-	powerByte := store.Get(types.GetDymintSeqKey(consAddr))
-	if powerByte == nil {
-		return 0, false
+	if operatorAddr.Empty() || consensusPubKey == nil {
```

Comment:
seems to me like this will always fail as always one of them will be true

Reply:
we expect 2 records, one `ed25519` and additional one